taniumrest_connector.py

```diff
@@ -940,94 +988,119 @@ def _handle_get_question_results(self, param):
-                          "question_text": query["question_text"]}
-        if "group" in query:
-            # Add in filters
-            question_data["group"] = query["group"]
-
         return question_data
 
+    def _create_sensor_dict(self, action_result, sensor, param_list):
+        sensor_name = sensor["name"]
+        endpoint = TANIUMREST_GET_SENSOR_BY_NAME.format(sensor_name=sensor_name)
+        ret_val, response = self._make_rest_call_helper(action_result, endpoint, verify=self._verify)
+        if phantom.is_fail(ret_val):
+            action_result.set_status(phantom.APP_ERROR, "Failed to get sensor definition from Tanium")
+            return phantom.APP_ERROR, {}
+
+        response_data = response.get("data")
+        if not response_data:
+            action_result.set_status(phantom.APP_ERROR, TANIUMREST_RESOURCE_NOT_EXIST.format(sensor_name, "sensor"))
+            return phantom.APP_ERROR, {}
+
+        resp_data = self._get_response_data(response_data, action_result, "sensor")
+        if resp_data is None:
+            return phantom.APP_ERROR, {}
+
+        self.save_progress("Parameter Definition:\n{}".format(resp_data.get("parameter_definition", "")))
+
+        raw_parameter_definition = resp_data.get("parameter_definition", "")
+        parameter_definition = None
+        try:
+            if raw_parameter_definition:
+                parameter_definition = json.loads(raw_parameter_definition)
+        except Exception as e:
+            error_code, error_msg = self._get_error_message_from_exception(e)
+            error_msg = "Error while parsing the 'parameter_definition'. Error Code: {0}. Error Message: {1}".format(
+                error_code, error_msg)
+            return action_result.set_status(phantom.APP_ERROR, error_msg), {}
+
+        param_count = 0
+        if parameter_definition:
+            # Parameterized Sensor
+            parameter_keys = [parameter["key"] for parameter in parameter_definition["parameters"]]
+            self.save_progress("Parameter Keys:\n{}".format(json.dumps(parameter_keys)))
+            param_count += len(parameter_keys)
+            parameters = []
+
+            for key in parameter_keys:
+                if self._param_idx >= len(param_list):
+                    action_result.set_status(phantom.APP_ERROR, TANIUMREST_NOT_ENOUGH_PARAMS)
+                    return
+
+                parameter = {
+                    "key": "||%s||" % key,
+                    "value": param_list[self._param_idx]
+                }
+                parameters.append(parameter)
+                self._param_idx += 1
+
+            sensor_dict = {
+                "source_hash": sensor["hash"],
+                "name": sensor_name,
+                "parameters": parameters
+            }
+            return phantom.APP_SUCCESS, sensor_dict
+        else:
+            # Regular Sensor, can use as-is
+            return phantom.APP_SUCCESS, sensor
```

Comment:
We don't need the `else` block here, right?

Reply:
Removed the else block and moved it to above as per the next suggestion.